### PR TITLE
Delegate to hydra node implementation

### LINKED ISSUE #158: Implement Hydra node requests on delegate server
Most of things should be already covered by #124 and #125 

Domain logic should be covered by other tasks.

Scope of this task is to actually react on `DelegateEvents` by triggering actions on L2. Concretelly:

- [ ] Introduce Hydra node and Cardano node connection in config or CLI option
- [x] Init/Abort/Close Head
- [x] Commit - both empty and StandingBid transactions case
- [x] Placing new bid
- [x] Probably testing this in E2E test?

No signing by bidder is required in this milestone.

Blocked by #140 
Blocked by #124 
Blocked by #125

### LINKED ISSUE #174: Trigger automatic actions defined by Hydra events in delegate server
This task requires to introduce new thread placing in queue HydraEvents which are triggered by:

* `DelegateEvent HydraEvent`

For that, probably, some new utilities for Hydra responses monitoring are required.

Actual implementation of responses to them is out of scope of this task.

Blocked by #125
Blocked by https://github.com/mlabs-haskell/hydra-auction/issues/140

### PR DESCRIPTION
**Pull request by uhbif19**
CLI parsing is very hacked now, will fix that in following PR.

Closes #158 
Closes #174 

Blocked by #259

### CODE REVIEW

File: delegate-app/Main.hs [hunk 1]
```diff
@@ -165,28 +175,31 @@ runDelegateLogicSteps ::
...
-    flushAndTraverseNewElements
-      frontendRequestQueue
-      ( delegateFrontendRequestStep
-          >=> putInToClientsQueue
-      )
-    -- FIXME: log queues overload and make tick not-static
-    liftIO $ threadDelay tick
+  broadcast = do
+    (voidResult, _) <-
+      flip runStateT initialState $ forever $ do
```
Comment (nazrhom) [inline delegate-app/Main.hs]: nit: you can use `evalState` here to avoid having to pattern match on void an return it

File: delegate-app/Main.hs [hunk 2]
```diff
@@ -306,23 +344,37 @@ mbQueueAuctionPhases delegateEvents toClientsChannel = do
...
-          { host
-          , port = fromMaybe portDefault $ port >>= readMaybe
-          , tick = tick
-          , ping = ping
-          }
-  runWithTracer' tracer $ runDelegateServer conf
+  -- FIXUP: parse actual adress and other params
+  hydraNodeNumber <- read . fromJust <$> lookupEnv "HYDRA_NODE_NUMBER"
+  putStrLn $ "With number: " <> show hydraNodeNumber
+  let actor = case hydraNodeNumber of
```
Comment (nazrhom) [inline delegate-app/Main.hs:352]: nit: I think it would be nicer to respect the natural ordering of the alphabet (oscar-patricia-rupert)

Comment (uhbif19) [reply delegate-app/Main.hs:352]: Yes, that is not only non-nice, but probably a reason for previous bug. I forgot to backport fix.

File: src/HydraAuction/OnChain/StandingBid.hs [hunk 3]
```diff
@@ -76,28 +75,30 @@ mkStandingBidValidator terms datum redeemer context =
...
-          (oldApprovedBidders == newApprovedBidders)
-        && case oldBid of
-          Just oldBidTerms ->
-            traceIfFalse "Bid increment is not greater than minimumBidIncrement" $
-              bidAmount oldBidTerms + minimumBidIncrement terms <= bidAmount newBidTerms
-          Nothing ->
-            traceIfFalse "Bid is not greater than startingBid" $
-              startingBid terms <= bidAmount newBidTerms
+    validNewBid (StandingBidState _oldApprovedBidders oldBid) (StandingBidState _newApprovedBidders (Just newBidTerms)) =
+      -- FIXME: disabled until M6
```
Comment (nazrhom) [inline src/HydraAuction/OnChain/StandingBid.hs:79]: Why do we need to remove this until m6? We should be able to make this work already

Comment (uhbif19) [reply src/HydraAuction/OnChain/StandingBid.hs:79]: This breaks, cuz delegate is signing Tx, while it is not in list of bidders. And knowing delegate list by Frontend CLI also requires new code.

And in M6 solution will be changed to completely different anyway.

**Review by nazrhom — APPROVED** [review]

File: delegate-app/Main.hs (hunk 2 above)
Comment (nazrhom) [reply delegate-app/Main.hs:352]: I noticed this is also wrong in the docker-compose, my bad :D

**Pull request merged**